### PR TITLE
Add test-msys2 job to circleci pipeline

### PR DESCRIPTION
**Pull request by mtmcp**
Run steps are modeled on the test-linux job but executed inside an MSYS2 shell.

### CODE REVIEW

**Comment by mmcparlane** [comment]
Hmm.. It's not showing the Windows tests on my side, and also not showing the newly added test-msys2. Perhaps due to my personal circleci not having credits... Can someone else verify these?

**Comment by kripken** [comment]
I think you need to also add the new job to the list of jobs to run, and not just define it. See the list near the end of the file, under `workflows:`

**Comment by mmcparlane** [comment]
Oops. Easy enough!

**Comment by mmcparlane** [comment]
Yikes. The `base-devel` installs way too many packages. I'll have to remove all but the essentials to run the basic test scripts...

**Comment by mmcparlane** [comment]
Anyone want to test this PR?

File: .circleci/config.yml [hunk 1]
```diff
@@ -92,6 +92,44 @@ jobs:
...
+          command: C:\msys64\usr\bin\bash -lc 'pacman --noconfirm --sync --needed msys2-devel python cmake'
+      - run:
+          name: test.sh
+          command: C:\msys64\usr\bin\bash -lc "export EMSDK_NOTTY='1' && ./scripts/test.sh"
+      - run:
+          name: test_source_env.sh
+          command: C:\msys64\usr\bin\bash -lc "export EMSDK_NOTTY='1' && ./scripts/test_source_env.sh"
+      - run:
+          name: test.py
+          command: C:\msys64\usr\bin\bash -lc "source emsdk_env.sh && export EMSDK_NOTTY='1' && /usr/bin/python scripts/test.py"
```
Comment (sbc100) [inline .circleci/config.yml:132]: BTW, although it would be nice to run emsdk itself under msys python, my current thinking is that we will require windows python when running emcc itself.

This is because emcc is a wrapper around a windows (not msys) version of llvm.   i.e,  I think its reasonable for emcc, just like llvm which is wraps to accept *only* valid windows paths as inputs.

Not that this work you are doing won't be useful as way to bootstrap emsdk which can then install windows python.

Comment (mmcparlane) [reply .circleci/config.yml:132]: It actually shouldn't matter which python is run because the next set of changes to get this working will be to update the python scripts to recognize (via environment variable) the msys/mingw environments so it can adjust the path separators properly. But it makes sense to use the system python as most users will have that anyhow—saves the CI system some cycles/bandwidth too! ⚡

Comment (sbc100) [reply .circleci/config.yml:132]: Do you mean the emscripten python scripts?

Do you imagine `emcc /mnt/c/path/to/src.cpp -L/mnt/c/lib`  working?    I think the amount of changes to emcc required to make that would be a lot.    Also.. even if you convert all those paths the windows paths that llvm/clang would need under the hood.. you than get clang error message that contains the windows paths ("C:\path\to\src") so now you need to think about capuring the clang output and do string replacements within it just to get reasonable errors messages... I think it will be long tail of annoying work..   it might be easier to instread require an msys version of llvm.. then everything should just work.

In summary:  I believe to get emcc working on msys2 you first need to get clang that works on msys2.

Comment (sbc100) [reply .circleci/config.yml:132]: Getting emsdk working on msys2 is different question.. we can support that without supporting msys paths in emcc.

Comment (mmcparlane) [reply .circleci/config.yml:132]: > In summary: I believe to get emcc working on msys2 you first need to get clang that works on msys2.

Ah, good point. Looks like I'll have to make a couple changes to use the mingw environment provided by msys2—prebuilt clang packages already provided:

```
mtmcp@JollyBox MSYS ~
$ pacman -Ss llvm
mingw32/mingw-w64-i686-clang 10.0.1-1
    C language family frontend for LLVM (mingw-w64)
mingw32/mingw-w64-i686-compiler-rt 10.0.1-1
    Runtime libraries for Clang and LLVM (mingw-w64)
mingw32/mingw-w64-i686-lld 10.0.1-1
    Linker tools for LLVM (mingw-w64)
mingw32/mingw-w64-i686-llvm 10.0.1-1
    Low Level Virtual Machine (mingw-w64)
mingw32/mingw-w64-i686-polly 10.0.1-1
    Polly - Polyhedral optimizations for LLVM (mingw-w64)
mingw64/mingw-w64-x86_64-clang 10.0.1-1
    C language family frontend for LLVM (mingw-w64)
mingw64/mingw-w64-x86_64-compiler-rt 10.0.1-1
    Runtime libraries for Clang and LLVM (mingw-w64)
mingw64/mingw-w64-x86_64-lld 10.0.1-1
    Linker tools for LLVM (mingw-w64)
mingw64/mingw-w64-x86_64-llvm 10.0.1-1
    Low Level Virtual Machine (mingw-w64)
mingw64/mingw-w64-x86_64-polly 10.0.1-1
    Polly - Polyhedral optimizations for LLVM (mingw-w64)
```

> Getting emsdk working on msys2 is different question.. we can support that without supporting msys paths in emcc.

Personally, I would probably just use Windows shells to pull down emsdk stuff (that way I could build on Windows as well). In that regard, what is the goal in terms of the test scripts? Do we want to be able to run all of these from msys2?

1. scripts/test.sh
2. scripts/test_source_env.sh
3. scripts/test.py
